kitsune/customercare/zendesk.py

```diff
@@ -74,6 +76,18 @@ def update_user(self, user):
         # https://developer.zendesk.com/api-reference/ticketing/users/user_identities/#make-identity-primary
         return zendesk_user
 
+    def get_primary_email_identity(self, zendesk_user_id):
+        """Update the primary email of the user."""
+
+        for identity in self.client.users.identities(id=zendesk_user_id):
+            if identity.primary and identity.type == "email":
+                return identity.id
+
+    def update_primary_email(self, zendesk_user_id, identity_id, email):
```

Comment:
This method also feels very disconnected from the `update_user` method above. Having this functionality, but not including it in the `update_user` method makes me think that method should really be called `update_user_except_for_primary_email`.

Additionally, `update_user` as written *will* update emails, but add them as secondaries. So it's entirely possible - likely even, given the order of when events are fired - that a user with an updated email address will first have their identities set like so, after `update_user` is called:
`[ { primary: true, value: "old@example.com" }, { primary: false, value: "new@example.com" } ]`
and then, after `update_primary_email` is called:
`[ { primary: true, value: "new@example.com" }, { primary: false, value: "new@example.com"} ]`
Zendesk may handle this gracefully and dedupe the emails, but I wouldn't want to count on it.

I think these new methods lend themselves more towards being private methods - with them taking an explicit `zendesk_user_id` and `email` string, rather simply a django user object. `update_user` could then live up to its name, and properly and completely update the user, working out what it needs to call in zendesk to do that.

Reply:
Indeed, this is exactly what happens. Updating one's primary email in FxA, then logging in results in:
```
APIException at /fxa/callback/

{"error": "RecordInvalid", "description": "Record validation errors", "details": {"value": [{"description": "Value: lmcardle+1-new-2@mozilla.com is already being used by another user", "error": "DuplicateValue"}]}}
```
Because `update_user` has already added the new email as a secondary.

Reply:
Per our discussion we are going to remove email updates from the update_user. Primary email will be updated only through the new method introduced here. TBD about alternate emails